### PR TITLE
Support gpu version crossvalidator for RandomForestRegressor

### PR DESCRIPTION
**Pull request by wbo4958**

### CODE REVIEW

**Comment by wbo4958** [comment]
build

File: python/src/spark_rapids_ml/regression.py [hunk 1]
```diff
@@ -833,3 +843,84 @@ def cpu(self) -> SparkRandomForestRegressionModel:
...
+    def _get_cuml_transform_func(
+        self, dataset: DataFrame, category: str = transform_evaluate.transform
+    ) -> Tuple[_ConstructFunc, _TransformFunc, Optional[_EvaluateFunc],]:
+        _construct_rf, _predict, _ = super()._get_cuml_transform_func(dataset, category)
+
+        def _evaluate(
+            input: TransformInputType,
+            transformed: TransformInputType,
+        ) -> pd.DataFrame:
+            # calculate the count of (label, prediction)
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/regression.py:856]: count -> metrics

**Review by eordentlich — COMMENTED** [review]
LGTM, minor comments

File: python/src/spark_rapids_ml/regression.py [hunk 2]
```diff
@@ -833,3 +843,84 @@ def cpu(self) -> SparkRandomForestRegressionModel:
...
+        dataset : :py:class:`pyspark.sql.DataFrame`
+            a dataset that contains labels/observations and predictions
+        evaluator: :py:class:`pyspark.ml.evaluation.Evaluator`
+            an evaluator user intends to use
+        params : dict, optional
+            an optional param map that overrides embedded params
+
+        Returns
+        -------
+        float
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py]: Good catch. Done

File: python/src/spark_rapids_ml/regression.py (hunk 1 above)
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py]: Good catch. Done

**Comment by wbo4958** [comment]
build

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**